src/index.js

```diff
@@ -1,37 +1,81 @@
-    index: 1,
-  },
-  {
-    description: 'Clear all completed',
-    completed: true,
-    index: 2,
-  },
-];
-
 const todoFullList = document.getElementById('todoFullList');
+const input = document.getElementById('input-type');
+const inputEnter = document.querySelector('.enterSng');
 
-const Display = () => {
-  Lists.forEach((e) => {
-    const li = document.createElement('li');
-    li.innerHTML = `<div class="inputDescripation"> <input type="checkbox" /> <span class="descriptions" >${e.description}</span> </div> <div class="bin">  </div>`;
-    li.setAttribute('key', e.index);
-    if (e.completed) {
-      li.classList.add('li');
-    }
-    todoFullList.appendChild(li);
-  });
-};
+let tasks = [];
+
+inputEnter.addEventListener('click', (() => {
+  if (input.value === '') {
+    alert('please type');
+  } else {
+    const newtask = {
+      description: input.value,
+      completed: false,
+      index: tasks.length,
+    };
+    tasks.push(newtask);
+    todoFullList.innerHTML = '';
+    tasks.forEach((e, index) => {
+      todoFullList.innerHTML += `
+          <div>
+            <li class="todo-child" id=${index}>
+              <div>
+              <input class="checkbox" type="checkbox" ${(e.completed) ? 'checked' : ''}>
+               <span> ${e.description} </span>
+              </div>
+              <div class=bin></div>
+            </li>
+          </div>
+        `;
+    });
+    input.value = '';
+    localStorage.setItem('todo-list', JSON.stringify(tasks));
+  }
+}));
+
+window.addEventListener('load', () => {
+  const getItem = JSON.parse(localStorage.getItem('todo-list'));
+  if (getItem) {
+    todoFullList.innerHTML = getItem.map((e, index) => `
+        <div>
+          <li class="todo-child" id=${index}>
+            <div>
+            <input class="checkbox" type="checkbox" ${(e.completed) ? 'checked' : ''}>
+            <span> ${e.description} </span>
+            </div>
+            <div class=bin></div>
+          </li>
+        </div>
+      `).join('');
+    tasks = getItem;
 
-document.addEventListener('DOMContentLoaded', () => {
-  Display();
+    document.querySelectorAll('.checkbox').forEach((checkbox, index) => {
+      checkbox.addEventListener('click', function () {
+        tasks[index].completed = this.checked;
+        localStorage.setItem('todo-list', JSON.stringify(tasks));
+      });
+    });
+  }
+});
+
+todoFullList.addEventListener('click', (e) => {
+  if (e.target.classList[0] === 'bin') {
+    e.target.parentElement.remove();
+    const newArray = tasks
+      .filter((element, index) => index !== parseInt(e.target.parentElement.id, 10))
+      .map((element, index) => {
+        element.index = index;
+        return element;
+      });
+    const li = document.querySelectorAll('.todo-child');
+    for (let i = 0; i < newArray.length; i += 1) {
+      newArray[i].index = i;
+      li[i].id = i;
+    }
+    tasks = newArray;
+    localStorage.setItem('todo-list', JSON.stringify(newArray));
```

Comment:
- Please ensure the property index is set to the value of the new array length (i.e., since I have a 5th task added to the list, the index of that task should equal 5). 👍🏽
![Screenshot from 2023-03-30 09-02-40](https://user-images.githubusercontent.com/44624138/228770685-d88244cf-b7da-4fcf-bddd-405469dbe7b9.png)

Reply:
- Please ensure a function for editing task descriptions is implemented as required for this milestone 😉
![Screenshot from 2023-03-29 09-44-52](https://user-images.githubusercontent.com/44624138/228478942-a99e17e2-cc51-4e41-8fad-871436a6f874.png)

- Please don't forget to create a new JavaScript file for the new functionality.
![Screenshot from 2023-03-30 09-06-35](https://user-images.githubusercontent.com/44624138/228771624-85ff4e7c-8b11-43a7-88c6-12113ebac858.png)